src/api/app/controllers/concerns/webui/notifications_filter.rb

```diff
@@ -1,33 +1,56 @@
 module Webui::NotificationsFilter
   extend ActiveSupport::Concern
 
-  # It's just a case...
+  # It's a sequence of multiple conditions combinations
   # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
   def filter_notifications_by_type(notifications, filter_type)
-    case filter_type
-    when 'comments'
-      notifications.for_comments
-    when 'requests'
-      notifications.for_requests
-    when 'incoming_requests'
-      notifications.for_incoming_requests(User.session!)
-    when 'outgoing_requests'
-      notifications.for_outgoing_requests(User.session!)
-    when 'relationships_created'
-      notifications.for_relationships_created
-    when 'relationships_deleted'
-      notifications.for_relationships_deleted
-    when 'build_failures'
-      notifications.for_failed_builds
-    when 'reports'
-      notifications.for_reports
-    when 'workflow_runs'
-      notifications.for_workflow_runs
-    when 'appealed_decisions'
-      notifications.for_appealed_decisions
-    else # all
-      notifications
-    end
+    return notifications if filter_type == ['all']
+
+    relations_type = []
+    relations_type << notifications.for_comments if filter_type.include?('comments')
+    relations_type << notifications.for_requests if filter_type.include?('requests')
+    relations_type << notifications.for_incoming_requests(User.session) if filter_type.include?('incoming_requests')
+    relations_type << notifications.for_outgoing_requests(User.session) if filter_type.include?('outgoing_requests')
+    relations_type << notifications.for_relationships_created if filter_type.include?('relationships_created')
+    relations_type << notifications.for_relationships_deleted if filter_type.include?('relationships_deleted')
+    relations_type << notifications.for_build_failures if filter_type.include?('build_failures')
+    relations_type << notifications.for_reports if filter_type.include?('reports')
+    relations_type << notifications.for_workflow_runs if filter_type.include?('workflow_runs')
+    relations_type << notifications.for_appealed_decisions if filter_type.include?('appealed_decisions')
+
+    notifications = notifications.merge(relations_type.inject(:or)) unless relations_type.empty?
+    notifications
   end
   # rubocop:enable Metrics/CyclomaticComplexity
+  # rubocop:enable Metrics/PerceivedComplexity
+
+  def filter_notifications_by_state(notifications, filter_state)
+    if filter_state.include?('unread') && filter_state.include?('read')
+      notifications.merge(notifications.unread.or(notifications.read))
+    elsif filter_state.include?('read')
+      notifications = notifications.read
+    else
+      notifications = notifications.unread
+    end
+    notifications
```

Comment:
```suggestion
    elsif filter_state.include?('read')
      notifications.read
    else
      notifications.unread
    end
```

I think you don't need to set `notifications` the `if` will return whatever you eval in each branch, and in line https://github.com/openSUSE/open-build-service/pull/16183/files#diff-b4fff3d37707c2902ce0bc23c055562446e786501533af15c8fb7b648d187631R74 you already set this to `@notifications`, so I think it's redundant.